src/access/examples/ExampleAccessControlDelegate.cpp

```diff
@@ -0,0 +1,1101 @@
+
+    static EntryDelegate * Find(Entry::Delegate & candidate)
+    {
+        if (InPool(candidate))
+        {
+            return &static_cast<EntryDelegate &>(candidate);
+        }
+        for (auto & delegate : pool)
+        {
+            if (!delegate.InUse())
+            {
+                return &delegate;
+            }
+        }
+        return nullptr;
+    }
+
+    static bool InPool(const Entry::Delegate & delegate)
+    {
+        constexpr auto end = pool + ArraySize(pool);
+        return pool <= &delegate && &delegate < end;
+    }
+
+public:
+    void Release() override
+    {
+        assert(mStorage != nullptr);
+        mStorage->Release();
+        mStorage = nullptr;
+    }
+
+    CHIP_ERROR GetAuthMode(AuthMode & authMode) const override
+    {
+        authMode = mStorage->mAuthMode;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR GetFabricIndex(FabricIndex & fabricIndex) const override
+    {
+        fabricIndex = mStorage->mFabricIndex;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR GetPrivilege(Privilege & privilege) const override
+    {
+        privilege = mStorage->mPrivilege;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR SetAuthMode(AuthMode authMode) override
+    {
+        ReturnErrorCodeIf(EnsureInPool(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        mStorage->mAuthMode = authMode;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR SetFabricIndex(FabricIndex fabricIndex) override
+    {
+        ReturnErrorCodeIf(EnsureInPool(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        mStorage->mFabricIndex = fabricIndex;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR SetPrivilege(Privilege privilege) override
+    {
+        ReturnErrorCodeIf(EnsureInPool(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        mStorage->mPrivilege = privilege;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR GetSubjectCount(size_t & count) const override
+    {
+        count = 0;
+        for (const auto & subject : mStorage->mSubjects)
+        {
+            if (subject.IsEmpty())
+            {
+                break;
+            }
+            count++;
+        }
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR GetSubject(size_t index, NodeId & subject) const override
+    {
+        if (index < EntryStorage::kMaxSubjects)
+        {
+            return mStorage->mSubjects[index].Get(subject);
+        }
+        return CHIP_ERROR_SENTINEL;
+    }
+
+    CHIP_ERROR SetSubject(size_t index, NodeId subject) override
+    {
+        ReturnErrorCodeIf(EnsureInPool(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        if (index < EntryStorage::kMaxSubjects)
+        {
+            return mStorage->mSubjects[index].Set(subject);
+        }
+        return CHIP_ERROR_SENTINEL;
+    }
+
+    CHIP_ERROR AddSubject(size_t * index, NodeId subject) override
+    {
+        ReturnErrorCodeIf(EnsureInPool(), CHIP_ERROR_BUFFER_TOO_SMALL);
+        size_t count;
+        GetSubjectCount(count);
```

Comment:
This looks side-effecting. Would look much more legible as:

```
size_t count = GetSubjectCount();
```

If keeping the "getter outputting to reference arg" pattern, please set the `count` to zero.

Reply:
I've set count to zero in all cases.